chain/pkg/gzip/gzip.go

```diff
@@ -0,0 +1,39 @@
+package gzip
+
+import (
+	"bytes"
+	gz "compress/gzip"
+	"errors"
+	"io"
+	"io/ioutil"
+)
+
+// magic bytes to identify gzip.
+// See https://www.ietf.org/rfc/rfc1952.txt
+// and https://github.com/golang/go/blob/master/src/net/http/sniff.go#L186
+var gzipIdent = []byte("\x1F\x8B\x08")
+
+// IsGzip returns true if the input file is gzipped file.
+func IsGzipped(src []byte) bool {
+	return bytes.Equal(gzipIdent, src[0:3])
+}
+
+// Uncompress returns gzip uncompressed.
```

Comment:
Uncompress returns uncompressed gzip, or error if input is not gzipped or too large.